// nlp-regex/index.js
/**
 * Measures of similarity between two texts.
 *
 * Jaro, Jaro-Winkler and Levenshtein distance
 * jDist, jwDist, lsDist
 */

rd = require('./distance.js');
st = require('./similar.js');

module.exports = {
  
  sText: function(str1,str2,percent){
    return st.similarText(str1,str2,percent);
  },
 
  jDist: function(str1,str2){
    return rd.jaroDistance(str1,str2);
  },

  jwDist: function(str1,str2){
    return rd.jwDistance(str1,str2);
  },

  lsDist: function(str1,str2){
    return rd.lsDistance(str1,str2);
  },

  /**
  * Escape special characters in the given string of html.
  *
  * @param  {String} html
  * @return {String}
  */
  escape: function(html) {
    return String(html)
      .replace(/&/g, '&amp;')
      .replace(/"/g, '&quot;')
      .replace(/'/g, '&#39;')
      .replace(/</g, '&lt;')
      .replace(/>/g, '&gt;');
  },

  /**
   * Unescape special characters in the given string of html.
   *
   * @param  {String} html
   * @return {String}
   */
  unescape: function(html) {
    return String(html)
      .replace(/&amp;/g, '&')
      .replace(/&quot;/g, '"')
      .replace(/&#39;/g, '\'')
      .replace(/&lt;/g, '<')
      .replace(/&gt;/g, '>');
  }
};
